src/zbeacon.c

```diff
@@ -337,6 +337,26 @@ s_self_prepare_udp (self_t *self)
                             (char *) &if_index, sizeof (if_index)))
                 zsys_socket_error (
                   "zbeacon: setsockopt IP_MULTICAST_IF failed");
+#else
```

Comment:
This is not quite right - it's losing IP_ADD_MEMBERSHIP on the sender socket and IP_MULTICAST_IF on the receiver socket. Instead of ifdeffing the whole thing, please restrict it to the IP_MULTICAST_IF calls.

Reply:
I've not tested it with IP_MULTICAST_IF on the receiver socket and IP_ADD_MEMBERSHIP on the sender.
OK, it needs to be checked on. So, this PR is closed.